### PR TITLE
Feature/azure cost collection

### PR DESCRIPTION
**Pull request by JohannesRudolph**
This PR fixes and simplifies Azure Cost collection as well as making configuration of management group based collection obsolete. See commit messages for description of rationale

### CODE REVIEW

File: src/deps.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export {
   existsSync,
 } from "https://deno.land/std@0.116.0/fs/mod.ts";
 export * as path from "https://deno.land/std@0.116.0/path/mod.ts";
+export * as async from "https://deno.land/std@0.116.0/async/mod.ts";
```
Comment (tfelix) [inline src/deps.ts:18]: interesting that this does not interfere with the async keyword.

**Review by tfelix — CHANGES_REQUESTED** [review]
Please have a look at this modified error handling. I think this might be an issue. The rest looks good.

File: src/deps.ts (hunk 1 above)
Comment (JohannesRudolph) [reply src/deps.ts:18]: right :-) I hope typescript got its preference rules down

File: src/azure/basic-azure-cli-facade.ts [hunk 2]
```diff
@@ -79,46 +75,46 @@ export class BasicAzureCliFacade implements AzureCliFacade {
...
-      ) {
-        console.error(
-          `Could not list tags for Subscription ${subscription.id}: Access was denied`,
-        );
+    const result = await this.shellRunner.run(command);
 
-        return Promise.resolve([]);
-      }
-    }
+    this.checkForErrors(result);
```
[the remark this answers is not in the document]
Comment (JohannesRudolph) [reply src/azure/basic-azure-cli-facade.ts:80]: shouldn't that behavior then be solved in the retry decorator (i.e. never retry an unauthorized call)? The other operations in the facade don't special case` AzureErrorCode.AZURE_UNAUTHORIZED` either, so I was going for consistency here. 

By fixing the filtering on subscriptions from the local AAD tenant see #83 I have not seen listTags unauthorized issues in my usage of collie at least

[the remark this answers is not in the document]
Comment (tfelix) [reply src/azure/basic-azure-cli-facade.ts:80]: It is not so easy. Azure for example throws sometimes a 401 "Cert not valid" and such an error is retryable. But I checked the code and it seems this is cought and put into a special kind of Exception that the retryable decorater uses. So this should be fine.

**Review by tfelix — APPROVED** [review]
lgtm

**Pull request merged**